From 9ab3314ce5092cf9ba2b86eab970d2e9dddc17ce Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Wed, 30 May 2018 09:46:03 +0100
Subject: [fix] hackerrank, 'sherlock and valid string', python port

-iterate the correct (derived) set
-need to extract and use both keys and values
---
 .../sherlock.and.valid.string.py                 | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/hackerrank/algorithms.-.strings.-.medium.-.sherlock.and.the.valid.string/sherlock.and.valid.string.py b/hackerrank/algorithms.-.strings.-.medium.-.sherlock.and.the.valid.string/sherlock.and.valid.string.py
index 3f07f13..fe52481 100644
--- a/hackerrank/algorithms.-.strings.-.medium.-.sherlock.and.the.valid.string/sherlock.and.valid.string.py
+++ b/hackerrank/algorithms.-.strings.-.medium.-.sherlock.and.the.valid.string/sherlock.and.valid.string.py
@@ -26,21 +26,25 @@ def isValid(s):
     n_diff_n_of_counts = len(freq_2)
 
     res = "YES"
-    if n_diff_counts >= 3:
+    if n_diff_n_of_counts >= 3:
         res = "NO"
-    elif n_diff_counts == 2:
+    elif n_diff_n_of_counts == 2:
         n_diff_counts_w_2_or_more_items = 0
-        for k, v in freq.items():
+        for k, v in freq_2.items():
+            #print("k: %s, v: %d" % (k, v))
             if v > 1:
                 n_diff_counts_w_2_or_more_items += 1
         if n_diff_counts_w_2_or_more_items > 1:
             res = "NO"
         else:
-            # two value groups, one has a single element, great, but if it's > +/- 1
+            # two count groups, one has a single element, great, but if it's > +/- 1
             # of the other, then, no.. unless it's only a single instance..
             # eg. aaaabbbbc
-            k1, v1 = freq[0]
-            k2, v2 = freq[1]
+            keys = list(freq_2.keys())
+            values = list(freq_2.values())
+            k1, v1 = keys[0], values[0]
+            k2, v2 = keys[1], values[1]
+            #print("k1: %s, v1: %d, k2: %s, v2: %d" % (k1, v1, k2, v2))
             if int(k1) > 1 and int(k2) > 1 and abs(v2 - v1) > 1:
                  res = "NO"
 
-- 
2.23.0

